include/openPMD/Streaming.hpp

```diff
@@ -0,0 +1,31 @@
+#pragma once
+
+#include <list>
+#include <string>
+#include <utility>
+#include <vector>
+
+#include "openPMD/Dataset.hpp"
+#include "openPMD/benchmark/mpi/BlockSlicer.hpp"
+#include <unordered_map>
+
+namespace openPMD
+{
+enum class AdvanceStatus : unsigned char
```

Comment:
Please document the purpose of each enum and each member in those enums with Doxygen. This is a good place to mention high-level concepts where they are used :)

Reply:
I could not think of a non-redundant documentation for `AdvanceMode::BEGINSTEP` and `AdvanceMode::ENDSTEP`, but I added the others.